lib/Service/ETLService.php

```diff
@@ -120,23 +120,18 @@ public function extractAndTransform(int $max = self::MAX_BATCH_SIZE, OutputInter
 			->where($select->expr()->eq('id', $select->createParameter('id')));
 
 		foreach ($this->getRaw($max, $output) as $row) {
-			$insert->setParameter('uid', $row['uid']);
-			$insert->setParameter('ip', $row['ip']);
-			$insert->setParameter('seen', 1, IQueryBuilder::PARAM_INT);
-			$insert->setParameter('ts', $row['created_at'], IQueryBuilder::PARAM_INT);
-
-			try {
+			$select
+				->setParameter('uid', $row['uid'])
+				->setParameter('ip', $row['ip']);
+			$result = $select->execute();
+			$existing = $result->fetchAll()[0];
```

Comment:
:warning: this will fetch more than one row at a time. But you only use the first one.

Actually, `->fetch` should be fine and only return `null` if there was no data that matches the query or the cursor was moved to the end of the current result set during iteration.

Reply:
I understand but, at least on my machine (PHP7.2FPM) the `fetch()` in conjunction with `is_null` was failing regardless if it had any results or not and in effect it tried to update non existing rows. I figured that if I use `fetchAll()` instead it works.

Reply:
After running tests and analysing others code, I found that there was `closeCursor()` statement
```suggestion
			$result->closeCursor();
```